nifi-nar-bundles/nifi-extension-utils/nifi-prometheus-utils/src/main/java/org/apache/nifi/prometheus/util/PrometheusMetricsUtil.java

```diff
@@ -362,4 +365,149 @@ public static CollectorRegistry createBulletinMetrics(BulletinMetricsRegistry bu
         bulletinMetricsRegistry.setDataPoint(1, "BULLETIN", instanceId, componentType, componentId, parentId, nodeAddress, category, sourceName, sourceId, level);
         return bulletinMetricsRegistry.getRegistry();
     }
+
+    public static void aggregatePercentUsed(final ProcessGroupStatus status, final Map<String, Double> aggregatedMetrics) {
+        status.getProcessGroupStatus().forEach((childGroupStatus) -> aggregatePercentUsed(childGroupStatus, aggregatedMetrics));
+
+        for (ConnectionStatus connectionStatus : status.getConnectionStatus()) {
+            final double percentUsedBytes = getUtilization(connectionStatus.getQueuedBytes(), connectionStatus.getBackPressureBytesThreshold());
+            final double percentUsedCount = getUtilization(connectionStatus.getQueuedCount(), connectionStatus.getBackPressureObjectThreshold());
+
+            determineMaxValueForPercentUsed(aggregatedMetrics,
+                    "nifi_percent_used_bytes",
+                    percentUsedBytes);
+
+            determineMaxValueForPercentUsed(aggregatedMetrics,
+                    "nifi_percent_used_count",
+                    percentUsedCount);
+
+            setBackpressure(aggregatedMetrics, percentUsedBytes, "bytesAtBackpressure");
+            setBackpressure(aggregatedMetrics, percentUsedCount, "countAtBackpressure");
+        }
+    }
+
+    public static void aggregateConnectionPredictionMetrics(final Map<String, Double> aggregatedMetrics, final Map<String, Long> predictions) {
+        determineMinValueForPredictions(aggregatedMetrics,
+                "nifi_time_to_bytes_backpressure_prediction",
+                predictions.get("timeToBytesBackpressureMillis"),
+                "bytesAtBackpressure");
+
+        determineMinValueForPredictions(aggregatedMetrics,
+                "nifi_time_to_count_backpressure_prediction",
+                predictions.get("timeToCountBackpressureMillis"),
+                "countAtBackpressure");
+    }
+
+    private static void setBackpressure(final Map<String, Double> aggregatedMetrics, final double percentUsed, final String atBackpressureKey) {
+        if (percentUsed >= 100) {
+            aggregatedMetrics.put(atBackpressureKey, 1.0);
+        } else if (!aggregatedMetrics.containsKey(atBackpressureKey)) {
+            aggregatedMetrics.put(atBackpressureKey, 0.0);
+        }
```

Comment:
What do you think about defining static constants for some of these values?  For example, `1.0` could be defined as `MAXIMUM_BACKPRESSURE`, and that value could be use elsewhere for evaluation.